src/extensions/hooks/useInstalledExtensionsData.tsx

```diff
@@ -0,0 +1,116 @@
+import {
+  getLatestFailedAttemptFromWebhooks,
+  LatestWebhookDeliveryWithMoment,
+} from "@dashboard/apps/components/AppAlerts/utils";
+import { AppPaths } from "@dashboard/apps/urls";
+import { useFlag } from "@dashboard/featureFlags";
+import {
+  AppTypeEnum,
+  useEventDeliveryLazyQuery,
+  useInstalledAppsListQuery,
+} from "@dashboard/graphql";
+import { useHasManagedAppsPermission } from "@dashboard/hooks/useHasManagedAppsPermission";
+import { fuzzySearch } from "@dashboard/misc";
+import { mapEdgesToItems } from "@dashboard/utils/maps";
+import { Skeleton } from "@saleor/macaw-ui-next";
+import React, { useEffect, useState } from "react";
+
+import { AppDisabledInfo } from "../components/AppDisabledInfo";
+import { FailedWebhookInfo } from "../components/FailedWebhookInfo";
+import { ViewDetailsActionButton } from "../components/ViewDetailsActionButton";
+
+interface UseInstalledExtensionsDataProps {
+  searchQuery: string;
+}
+
+export const useInstalledExtensionsData = ({ searchQuery }: UseInstalledExtensionsDataProps) => {
+  const [initialLoading, setInitialLoading] = useState(true);
+  const { hasManagedAppsPermission } = useHasManagedAppsPermission();
+  const { enabled: appAlertsEnabled } = useFlag("app_alerts");
+
+  const { data, refetch } = useInstalledAppsListQuery({
+    variables: {
+      first: 100,
+      filter: {
+        type: AppTypeEnum.THIRDPARTY,
+      },
+      after: null,
+    },
+  });
+  const installedAppsData = mapEdgesToItems(data?.apps) || [];
+
+  const [queryEventDeliveries, { data: eventDeliveriesData, loading: eventDeliveriesLoading }] =
+    useEventDeliveryLazyQuery({
+      variables: {
+        first: 100,
+        filter: {
+          type: AppTypeEnum.THIRDPARTY,
+        },
+        canFetchAppEvents: hasManagedAppsPermission && appAlertsEnabled,
+      },
+    });
+  const eventDeliveries = mapEdgesToItems(eventDeliveriesData?.apps) ?? [];
```

Comment:
suggestion: I think after we fetch event deliveries we could create `Map` in order to make querying app problems later in this hook faster, so in this case we would have `key: <app id`, `value: event deliveries`. This way when creating `const installedApps` we won't have to iterate over entire list